src/main/java/nextstep/courses/domain/SessionType.java

```diff
@@ -0,0 +1,55 @@
+package nextstep.courses.domain;
+
+import java.util.function.BiConsumer;
+
+public enum SessionType {
+    PAID((price, maxStudent) -> {
+        validPaidSessionPriceIsNotZeroOrNegative(price);
+        validMaxStudentOfPaidSession(maxStudent);
+    }),
+    FREE((price, maxStudent) -> {
+        validFreeSessionIsFree(price);
+        validMaxStudentIsIntegerMax(maxStudent);
+    });
+
+    private final BiConsumer validator;
```

Comment:
제네릭 타입을 명시해주시면 좋을 것 같습니다.
```suggestion
    private final BiConsumer<Long, Integer> validator;
```